### PR TITLE
feat: add validation that `AIRFLOW__WEBSERVER__SECRET_KEY` is set when web is replicated

### LINKED ISSUE #265: fail if `AIRFLOW__WEBSERVER__SECRET_KEY` is not set with `web.replicas > 1`
Currently, if a user sets `web.replicas > 1` and forgets to set `AIRFLOW__WEBSERVER__SECRET_KEY`, then they will get a `BAD REQUEST: The CSRF session token is missing`.

We should add a `_helpers/validate-values.tpl`, which fails if the user forgets to set `AIRFLOW__WEBSERVER__SECRET_KEY`.  The issue is that there are multiple places where `AIRFLOW__WEBSERVER__SECRET_KEY` can be set:
1. `airflow.config.AIRFLOW__WEBSERVER__SECRET_KEY`
2. `airflow.extraEnv[]`.

Its easy to check if `airflow.config.AIRFLOW__WEBSERVER__SECRET_KEY` is set, but I am not sure that it's possible to check `airflow.extraEnv[]` contains an object with `name = AIRFLOW__WEBSERVER__SECRET_KEY` using helm templates.

### PR DESCRIPTION
**Pull request by henryzhangsta**
**What issues does your PR fix?**

- fixes #265 

**What does your PR do?**

- Adds validation that `AIRFLOW__WEBSERVER__SECRET_KEY` is set when `web.replicas` is greater than `1`.

## Checklist

- [x] Commits are [signed](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#sign-your-work)
- [x] Commits are [squashed](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#squash-commits) (only do this if appropriate)
- [ ] Chart [version bumped](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#versioning) (only do this if releasing a new version)
- [x] Chart [documentation updated](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#documentation)
- [x] Passes [linting](https://github.com/airflow-helm/charts/tree/main/charts/airflow/CONTRIBUTING.md#linting)

### CODE REVIEW

File: charts/airflow/templates/_helpers/validate-values.tpl [hunk 1]
```diff
@@ -61,6 +61,21 @@
...
+{{- if gt .Values.web.replicas 1. }}
+  {{- if not .Values.airflow.config.AIRFLOW__WEBSERVER__SECRET_KEY }}
+    {{- $foundKey := false }}
+    {{- range $env := .Values.airflow.extraEnv }}
+      {{- if eq $env.name "AIRFLOW__WEBSERVER__SECRET_KEY" }}
+        {{- $foundKey = true }}
+      {{- end }}
+    {{- end }}
+    {{- if not $foundKey }}
+      {{ required "AIRFLOW__WEBSERVER__SECRET_KEY is required to be set when there is more than one web replica" nil }}
```
Comment (thesuperzapper) [inline charts/airflow/templates/_helpers/validate-values.tpl:74]: Can you change the error message to be more consistent with the rest, for example:
```
If `web.replicas > 1`, you must define `AIRFLOW__WEBSERVER__SECRET_KEY`!
```

File: charts/airflow/values.yaml [hunk 2]
```diff
@@ -556,6 +556,7 @@ web:
   ##
   ## NOTE:
   ## - if you set this >1 we recommend defining a `web.podDisruptionBudget`
+  ## - if this is greater than one, then `AIRFLOW__WEBSERVER__SECRET_KEY` is required to be configured
```
Comment (thesuperzapper) [inline charts/airflow/values.yaml:559]: To make it consistent with the line above, can you make the string:
```
  ## the number of web Pods to run
  ##
  ## WARNING:
  ##  - if set greater than `1`, you MUST set `AIRFLOW__WEBSERVER__SECRET_KEY`,
  ##    this prevents "CSRF session token is missing" errors when users are served by multiple web pods
  ##
  ## NOTE:
  ## - if set greater than `1`, we recommend defining a `web.podDisruptionBudget`
  ##
```

**Review by thesuperzapper — CHANGES_REQUESTED** [review]
@henryzhangsta I love this approach!

I never thought of using a temporary `$foundKey` variable to loop through the `extraEnv`!

**Comment by henryzhangsta** [comment]
@thesuperzapper updated

**Comment by thesuperzapper** [comment]
@henryzhangsta sorry to do this, but this PR is no longer needed, this is because as of airflow `2.1.2`, EVERY INSTALL needs to set `AIRFLOW__WEBSERVER__SECRET_KEY` (see issue https://github.com/airflow-helm/charts/issues/327). Therefore, I have made PR https://github.com/airflow-helm/charts/pull/346, which adds `airflow.webserverSecretKey` to ensure that `AIRFLOW__WEBSERVER__SECRET_KEY` is always set.

However, I plan to implement your extremely clever method to check the various places which `AIRFLOW__WEBSERVER__SECRET_KEY` in solving https://github.com/airflow-helm/charts/issues/333 (warning the user to set a non-default value).

Please don't let this discourage you from contributing to the chart, you have been awsome!

**Pull request closed without merging**